pkg/builder/devcontainer.go

```diff
@@ -0,0 +1,384 @@
+	if b.project.Build.Devcontainer.DevContainerFilePath != "" {
+		cmd = append(cmd, "--config", filepath.Join("/project", b.project.Build.Devcontainer.DevContainerFilePath))
+	}
+
+	execConfig := types.ExecConfig{
+		AttachStdout: true,
+		AttachStderr: true,
+		Cmd:          cmd,
+		Tty:          true,
+	}
+
+	r, w := io.Pipe()
+	var buildOutcome BuildOutcome
+
+	go func(buildOutcome *BuildOutcome) {
+		var lastLine string
+		scanner := bufio.NewScanner(r)
+		for scanner.Scan() {
+			lastLine = scanner.Text()
+			projectLogger.Write([]byte(lastLine + "\n"))
+
+			if strings.Contains(lastLine, `{"outcome"`) {
+				start := strings.Index(lastLine, "{")
+				end := strings.LastIndex(lastLine, "}")
+				if start != -1 && end != -1 && start < end {
+					lastLine = lastLine[start : end+1]
+				}
+
+				err = json.Unmarshal([]byte(lastLine), &buildOutcome)
+				if err != nil {
+					//	todo: handle properly
+					log.Error(err)
+				}
+			}
+		}
+		if err := scanner.Err(); err != nil {
+			//	todo: handle properly
+			log.Error(err)
+		}
+	}(&buildOutcome)
+
+	result, err := dockerClient.ExecSync(b.buildId, execConfig, w)
+	if err != nil {
+		return err
+	}
+
+	if buildOutcome.Outcome != "success" {
+		//	todo: parse reason
+		return errors.New("devcontainer build failed")
+	}
+
+	b.user = buildOutcome.RemoteUser
+
+	if result.ExitCode != 0 {
+		return errors.New("devcontainer build failed")
+	}
+
+	builderCli, err := b.getBuilderDockerClient()
+	if err != nil {
+		return err
+	}
+
+	//	todo: commit sha from project Repository
+	tag := "latest"
+	imageName := fmt.Sprintf("%s:%s", b.localContainerRegistryServer+"/p-"+b.buildId, tag)
+
+	_, err = builderCli.ContainerCommit(context.Background(), buildOutcome.ContainerId, container.CommitOptions{
+		Reference: imageName,
+	})
+	if err != nil {
+		return err
+	}
+
+	b.buildImageName = imageName
+
+	return nil
+}
+
+func (b *DevcontainerBuilder) readConfiguration() error {
+	projectLogger := b.loggerFactory.CreateProjectLogger(b.project.WorkspaceId, b.project.Name)
+	defer projectLogger.Close()
+
+	cli, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
+	if err != nil {
+		return err
+	}
+
+	dockerClient := docker.NewDockerClient(docker.DockerClientConfig{
+		ApiClient: cli,
+	})
+
+	cmd := []string{"devcontainer", "read-configuration", "--include-features-configuration", "--include-merged-configuration", "--workspace-folder", "/project"}
+	if b.project.Build.Devcontainer.DevContainerFilePath != "" {
+		cmd = append(cmd, "--config", filepath.Join("/project", b.project.Build.Devcontainer.DevContainerFilePath))
+	}
+
+	execConfig := types.ExecConfig{
+		AttachStdout: true,
+		AttachStderr: true,
+		Cmd:          cmd,
+		Tty:          true,
+	}
+
+	result, err := dockerClient.ExecSync(b.buildId, execConfig, nil)
+	if err != nil {
+		return err
+	}
+
+	if result.ExitCode != 0 {
+		return errors.New("devcontainer read configuration failed")
```

Comment:
We should at least log `result.Stderr`.

Reply:
If, for example, you manually select the devcontainer builder in the advanced config and provide an invalid path to the devcontainer file, you get this:
![Screenshot 2024-05-22 at 11 15 23](https://github.com/daytonaio/daytona/assets/26512078/a2528e20-e4f5-4346-a19b-cc985286df62)

The user should be aware of what was the issue with the build otherwise this will be hard to debug.